lms/djangoapps/verify_student/views.py

```diff
@@ -492,6 +487,13 @@ def _redirect_if_necessary(
                     url = reverse('verify_student_verify_now', kwargs=course_kwargs)
             else:
                 url = reverse('verify_student_start_flow', kwargs=course_kwargs)
+        elif message in [self.FIRST_TIME_VERIFY_MSG, self.UPGRADE_MSG]:
```

Comment:
I've moved the logic to it's own if-block. Now if a user is active and goes to purchase/upgrade to verified certificate and the ecommerce checkout is enabled, the user get's redirected to the ecommerce basket.